Cargo.toml

```diff
@@ -4,22 +4,19 @@ version = "0.1.0"
 authors = ["Samuel Thomas <sgt43@cornell.edu>", "Kenneth Fang <kwf37@cornell.edu>"]
 edition = "2018"
 build = false
-# build = "build.rs"
 
 [dependencies]
 structopt = "~0.3"
 quickcheck = "0.9.0"
 petgraph = "~0.4"
 better-panic = "~0.2"
 sexpy = "~0.5"
-itertools = "~0.7"
+itertools = "~0.9"
 bumpalo = "3.2.0"
 atty = "~0.2"
+strum = "0.18.0"
+strum_macros = "0.18.0"
```

Comment:
What are these for?

Reply:
These are for enumerating the `BackendOpt` variants so that we can generate a list of backend names from the enum definition. This just means that we don't have to maintain separate lists for valid backends and help messages. It's used in `cmdline.rs:55`

Reply:
You could make that a list to backends instead and then not have to do this. It’ll also allow you to call get_name and get the name of the backend.

More generally, i dislike relying on too many libraries. It makes things like building a web demo in the future harder since you don’t know if they can compile to web targets.

Reply:
You're probably right about this. I've rewritten it to avoid this dependency.